hydra-plutus/src/Hydra/Contract/Head.hs

```
@@ -350,6 +359,23 @@ checkContest ctx@ScriptContext{scriptContextTxInfo} contestationDeadline parties
       UpperBound (Finite time) _ ->
         traceIfFalse "upper bound beyond contestation deadline" $ time <= contestationDeadline
       _ -> traceError "contest: no upper bound defined"
+
+  ScriptContext{scriptContextTxInfo = txInfo} = ctx
+
+  contester = maybeToList $
+    case txInfoSignatories txInfo of
+      [signer] -> Just signer
+      _ -> Nothing
```

Should: fail early here using a `traceError` instead of using `Maybe`